Update how to document changes in contribution guide & PR template (#7267)
scikit-image/scikit-image: .github/PULL_REQUEST_TEMPLATE.md

## Release note

We use [changelist](https://github.com/scientific-python/changelist) to

> compile each pull request into an item in our release notes. Please refer to

Reviewer: ```suggestion
list each pull request in the release notes. Please refer to
```

Author: "list" directly after "changelist" seems somewhat redundant.

What do you think about

> We use changelist to describe changes in each pull request in our release notes.

Reviewer: Maybe it's just me and 'compile' is actually fine: I tend to think of the computing meaning *first* but really that's the *second* meaning of 'compile!'

Otherwise:

> We use changelist to log each pull request and its description in the release notes.

I find '*our* release notes' overly possessive :sweat_smile:

> compile each pull request into an item in our release notes. Please refer to

Reviewer: ```suggestion
compile each pull request into an item of the release notes. Please refer to
```

> [documenting changes](https://scikit-image.org/docs/stable/development/contribute.html#documenting-changes) and past [release notes](https://scikit-image.org/docs/stable/release_notes/index.html) for guidance and examples.

Reviewer: ```suggestion
[instructions](https://scikit-image.org/docs/stable/development/contribute.html#documenting-changes) and past [release notes](https://scikit-image.org/docs/stable/release_notes/index.html) for guidance and examples.
```
(they are "instructions [on how to document changes]")
